src/core/or/policies.c

```diff
@@ -990,6 +991,60 @@ fascist_firewall_choose_address_rs(const routerstatus_t *rs,
   }
 }
 
+/** Like fascist_firewall_choose_address_base(), but takes in a smartlist
+ * <b>lspecs</b> consisting of a link specifier.
+ */
+void
+fascist_firewall_choose_address_ls(const smartlist_t *lspecs,
+                                   firewall_connection_t fw_connection,
```

Comment:
Link specifiers are only used for ORPorts, so fw_connection is always FIREWALL_OR_CONNECTION.